examples/rotation_bin/rotation_bin/imp.rs

```diff
@@ -0,0 +1,203 @@
+            }
+            _ => unimplemented!(),
+        }
+    }
+
+    fn dispose(&self, _obj: &Self::Type) {
+        if let Some(child) = self.child.borrow_mut().take() {
+            child.unparent();
+        }
+    }
+}
+
+impl WidgetImpl for RotationBin {
+    fn request_mode(&self, _widget: &Self::Type) -> SizeRequestMode {
+        if self.child.borrow().is_none() {
+            return SizeRequestMode::ConstantSize;
+        };
+
+        match self.rotation.get() {
+            Rotation::Normal | Rotation::Deg180 => SizeRequestMode::HeightForWidth,
+            Rotation::Deg90 | Rotation::Deg270 => SizeRequestMode::WidthForHeight,
+        }
+    }
+
+    fn compute_expand(&self, _widget: &Self::Type, hexpand: &mut bool, vexpand: &mut bool) {
+        let child = match self.child.borrow().as_ref() {
+            None => {
+                *hexpand = false;
+                *vexpand = false;
+                return;
+            }
+            Some(c) => c.clone(),
+        };
+
+        match self.rotation.get() {
+            Rotation::Normal | Rotation::Deg180 => {
+                *hexpand = child.compute_expand(gtk::Orientation::Vertical);
+                *vexpand = child.compute_expand(gtk::Orientation::Horizontal);
+            }
+            Rotation::Deg90 | Rotation::Deg270 => {
+                *hexpand = child.compute_expand(gtk::Orientation::Horizontal);
+                *vexpand = child.compute_expand(gtk::Orientation::Vertical);
+            }
+        };
+    }
+
+    fn measure(
+        &self,
+        _widget: &Self::Type,
+        orientation: gtk::Orientation,
+        for_size: i32,
+    ) -> (i32, i32, i32, i32) {
+        if self.child.borrow().is_none() {
+            return (0, 0, -1, -1);
+        }
+
+        let orientation = match self.rotation.get() {
+            Rotation::Deg90 | Rotation::Deg270 => {
+                if orientation == gtk::Orientation::Horizontal {
+                    gtk::Orientation::Vertical
+                } else {
+                    gtk::Orientation::Horizontal
+                }
+            }
+            _ => orientation,
+        };
+
+        self.child
+            .borrow()
+            .as_ref()
+            .unwrap()
+            .measure(orientation, for_size)
+    }
+
+    fn size_allocate(&self, _widget: &Self::Type, width: i32, height: i32, baseline: i32) {
+        if self.child.borrow().is_none() {
+            return;
+        }
+
+        let (w, h) = match self.rotation.get() {
+            Rotation::Deg90 | Rotation::Deg270 => (height, width),
+            _ => (width, height),
+        };
+
+        let transform = gsk::Transform::new();
+        let transform = transform
+            .translate(&Point::new(width as f32 / 2.0, height as f32 / 2.0))
+            .unwrap();
+        let transform = transform.rotate(self.rotation.get().into()).unwrap();
+        let transform = transform.translate(&Point::new(-w as f32 / 2.0, -h as f32 / 2.0));
```

Comment:
Especially this virtual method could use some comments, but also the others to explain why and what they do